File: public/static/templates/spring-boot/src/main/java/com/eventstorming/external/Command.java
forEach: RelationCommandInfo
fileName: {{commandValue.namePascalCase}}Command.java
path: {{boundedContext.name}}/{{{options.packagePath}}}/external
except: {{#except commandValue}}{{/except}}
---
package {{options.package}}.external;

import javax.persistence.*;
import java.util.List;
import java.util.Date;
import lombok.Data;

@Data
public class {{commandValue.namePascalCase}}Command {

{{#commandValue.fieldDescriptors}}
    {{#isKey}}
    @Id
    {{/isKey}}
    private {{{className}}} {{nameCamelCase}};
{{/commandValue.fieldDescriptors}}
}

<function>
window.$HandleBars.registerHelper('except', function (command) {
    if(command._type.endsWith('Command')) {
        return (command.fieldDescriptors && command.fieldDescriptors.length == 0);
    }
    return true;
});
</function>

File: public/static/templates/go/domain/Command.go
forEach: Command
representativeFor: Command
fileName: {{namePascalCase}}Command.go
path: {{boundedContext.name}}/{{boundedContext.name}}
except: {{#except fieldDescriptors}}{{/except}}
---
package {{boundedContext.name}}

type {{namePascalCase}}Command struct{
{{#fieldDescriptors}}
    {{#isKey}}
    {{namePascalCase}} {{#typeCheck className}} {{/typeCheck}} `gorm:"primaryKey" json:"{{nameCamelCase}}" type:"{{#typeCheck className}} {{/typeCheck}}"`
    {{/isKey}}
    {{^isKey}}
    {{namePascalCase}} {{#typeCheck className}} {{/typeCheck}} `json:"{{nameCamelCase}}" type:"{{#typeCheck className}}{{/typeCheck}}"`
    {{/isKey}}
{{/fieldDescriptors}}
}

<function>
window.$HandleBars.registerHelper('except', function (fieldDescriptors) {
    return (fieldDescriptors && fieldDescriptors.length == 0);
});

 window.$HandleBars.registerHelper('typeCheck', function (className) {
        if(className.endsWith("String")){
            return "string"
        }
		else if(className.endsWith("Integer")){
			return "int"
		}
		else if(className.endsWith("Float")){
			return "float64"
		}
		else if(className.endsWith("Long")){
			return "int"
		}
		else if(className.endsWith("Boolean")){
			return "bool"
		}
		else if(className.endsWith("Double")){
			return "int"
		}

    });
</function>
